### ISSUE TITLE
Prometheus: Backend refactor

### ISSUE THREAD

**Issue by aocenas** [grafana/grafana#51639]
Over time the Prometheus backend changed a lot since [initial migration](https://github.com/grafana/grafana/issues/37784). There were also some SDK changes that allow us to write some things in easier way. So few problems that should be solved:
- [ ] There are 2 different clients living in parallel (see https://github.com/grafana/grafana/pull/48477)
- [x] Duplicated HTTP setup code due to 2 clients. https://github.com/grafana/grafana/pull/51061 https://github.com/grafana/grafana/pull/51436
- [x] We handle custom headers manually instead of leaving that to the SDK and infra code. https://github.com/grafana/grafana/pull/51518
- [x] Additional HTTP client caching due to how custom headers are handled. https://github.com/grafana/grafana/pull/51061 https://github.com/grafana/grafana/pull/51436
- [x] Resource calls are trying to handle the response instead of just proxying the req/res. https://github.com/grafana/grafana/pull/51436

**Comment by szabalza** [comment]
@aocenas Please create a new ticket for tasks that will not be covered here and still have to be done

**Comment by aocenas** [comment]
Closing this and moving the remaining tasks to https://github.com/grafana/grafana/issues/52738
